lti_tool/views.py

```diff
@@ -0,0 +1,309 @@
+from django.http import HttpResponse
+from django.views.decorators.clickjacking import xframe_options_exempt
+from django.views.decorators.csrf import csrf_exempt
+from django.shortcuts import redirect
+from django.contrib.auth.models import User
+from django.contrib.auth import login
+from course.views import InstanceView, ModuleView
+from course.models import Enrollment
+from pylti1p3.contrib.django import DjangoOIDCLogin, DjangoMessageLaunch
+from pylti1p3.deep_link_resource import DeepLinkResource
+from pylti1p3.lineitem import LineItem
+from django.urls import reverse
+from exercise.viewbase import ExerciseBaseView
+from exercise.models import LearningObject, BaseExercise
+from exercise.views import ExerciseView, SubmissionView
+from userprofile.models import UserProfile
+from course.models import CourseInstance, Enrollment, CourseModule
+from course.viewbase import CourseInstanceBaseView, CourseModuleBaseView
+from lib.viewbase import BaseTemplateView, BaseRedirectView, BaseMixin
+from authorization.permissions import ACCESS
+from .utils import get_tool_conf, get_launch_data_storage, get_launch_url
+
+@csrf_exempt
```

Comment:
These exempts could be supplied with a comment explaining why.  There seems to be others also later in this file, so a general comment in the beginning is probably good enough.